probes/hasDangerousWorkflowScriptInjection/patch/impl_test.go

```diff
@@ -13,3 +13,107 @@
 // limitations under the License.
 
 package patch
+
+import (
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/ossf/scorecard/v4/checker"
+)
+
+func Test_GeneratePatch(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name             string
+		inputFilepath    string
+		expectedFilepath string
+		// err      error
+	}{
+		// Extracted from real Angular fix: https://github.com/angular/angular/pull/51026/files
+		{
+			name:             "Real Example 1",
+			inputFilepath:    "realExample1.yaml",
+			expectedFilepath: "realExample1_fixed.yaml",
+		},
+		// Inspired on a real fix: https://github.com/googleapis/google-cloud-go/pull/9011/files
+		{
+			name:             "Real Example 2",
+			inputFilepath:    "realExample2.yaml",
+			expectedFilepath: "realExample2_fixed.yaml",
+		},
+		// Inspired from a real lit/lit fix: https://github.com/lit/lit/pull/3669/files
+		{
+			name:             "Real Example 3",
+			inputFilepath:    "realExample3.yaml",
+			expectedFilepath: "realExample3_fixed.yaml",
+		},
+		{
+			name:             "Test all (or most) types of user input that should be detected",
+			inputFilepath:    "allKindsOfUserInput.yaml",
+			expectedFilepath: "allKindsOfUserInput_fixed.yaml",
+		},
+		{
+			name:             "User's input is assigned to a variable before used",
+			inputFilepath:    "userInputAssignedToVariable.yaml",
+			expectedFilepath: "userInputAssignedToVariable_fixed.yaml",
+		},
+		{
+			name:             "Two incidences in different jobs",
+			inputFilepath:    "twoInjectionsDifferentJobs.yaml",
+			expectedFilepath: "twoInjectionsDifferentJobs_fixed.yaml",
+		},
+		{
+			name:             "Two incidences in same job",
+			inputFilepath:    "twoInjectionsSameJob.yaml",
+			expectedFilepath: "twoInjectionsSameJob_fixed.yaml",
+		},
+		{
+			name:             "Two incidences in same step",
+			inputFilepath:    "twoInjectionsSameStep.yaml",
+			expectedFilepath: "twoInjectionsSameStep_fixed.yaml",
+		},
+		{
+			name:             "Workflow already has env vars defined in different scopes",
+			inputFilepath:    "envVarsAlreadyExist.yaml",
+			expectedFilepath: "envVarsAlreadyExist_fixed.yaml",
+		},
+		{
+			name:             "Bad indentation is kept the same",
+			inputFilepath:    "badIndentationMultipleInjections.yaml",
+			expectedFilepath: "badIndentationMultipleInjections_fixed.yaml",
+		},
+		{
+			// Currently we're not keeping this pattern, as we always add a blankline after the env block
+			name:             "File with no blank lines between blocks",
+			inputFilepath:    "noLineBreaksBetweenBlocks.yaml",
+			expectedFilepath: "noLineBreaksBetweenBlocks_fixed.yaml",
+		},
```

Comment:
In such cases, where we know the current behavior isn't satisfactory, we can have two tests: one with what it SHOULD be (commented out), and one with how it is:

```suggestion
		{
			// Currently we're not keeping this pattern, as we always add a blankline after the env block
			name:             "File with no blank lines between blocks",
			inputFilepath:    "noLineBreaksBetweenBlocks.yaml",
			expectedFilepath: "noLineBreaksBetweenBlocks_fixed.yaml",
		},
		// {
			// Test that currently fails due to lack of style awareness.
			// Once proper behavior is implemented, enable this test
			// and delete the one above.
			// name:             "File with no blank lines between blocks",
			// inputFilepath:    "noLineBreaksBetweenBlocks.yaml",
			// expectedFilepath: "noLineBreaksBetweenBlocks_fixed.yaml",
		// },
```

Reply:
I like the idea of having the commented one and I'm doing that, but is the "current" one really necessary? Theoretically I think it doesn't test anything that was not covered on other cases. And we would already have the commented one to raise awareness on how it should be -- I can also mention on the comment that we currently always add a blank line

Reply:
Well, having the "current" test enabled ensures there are no regressions. But I agree, if the test doesn't test anything that's not covered by other tests already, it can be dropped.